BuildLibs/buildmap.py

```diff
@@ -443,90 +697,165 @@ def GetContainingSectorNearby(self, sector:int, point) -> Union[int,None]:
+
+    HEADER_SIZE = 43
+
+    def CreateHeaderPacker(self):
+        self.headerPacker = FancyPacker(
+            '<',
+            OrderedDict(
+                sig='4s',
+                version='h',
+                startPos='iii',
+                startAngle='h',
+                startSect='h',
+                pskybits='h',
+                visibility='i',
+                songid='i',
+                parallaxtype='c',
+                mapRevision='i',
+                num_sectors='H',
+                num_walls='H',
+                num_sprites='H',
+            )
         )
-        self.headerLen = 43
-        self.__dict__.update(self.headerPacker.unpack(data[:self.headerLen]))
+
+    def ReadHeaders(self) -> int:
+        super().ReadHeaders()
 
         if self.songid != 0 and self.songid != 0x7474614d and self.songid != 0x4d617474:
-            header = data[:self.headerLen]
-            header[6:self.headerLen] = MapCrypt(header[6:self.headerLen], 0x7474614d)
+            header = self.data[:self.HEADER_SIZE]
+            header[6:self.HEADER_SIZE] = MapCrypt(header[6:self.HEADER_SIZE], 0x7474614d)
             self.crypt = 1
-            self.__dict__.update(self.headerPacker.unpack(header[:self.headerLen]))
+            self.__dict__.update(self.headerPacker.unpack(header[:self.HEADER_SIZE]))
 
         self.exactVersion = self.version
         self.version = self.exactVersion >> 8
 
         if self.version == 7: # if (byte_1A76C8)
             self.header2Len = 128
-            header2Start = self.headerLen
-            header2data = data[header2Start:header2Start + 128]
+            header2Start = self.HEADER_SIZE
+            header2data = self.data[header2Start:header2Start + 128]
             if self.crypt:
                 header2data = MapCrypt(header2data, self.num_walls)
 
-            header2Packer = FancyPacker('<', OrderedDict(x_sprite_size='i', x_wall_size='i', x_sector_size='i'))
+            header2Packer = FancyPacker('<', OrderedDict(X_SPRITE_SIZE='i', X_WALL_SIZE='i', X_SECTOR_SIZE='i'))
             header2 = header2Packer.unpack(header2data[64:76]) # skip the 64 bytes starting padding
             self.__dict__.update(header2)
         else:
             self.header2Len = 0
-            self.x_sector_size = 60
-            self.x_sprite_size = 56
-            self.x_wall_size = 24
+            self.X_SECTOR_SIZE = 60
+            self.X_SPRITE_SIZE = 56
+            self.X_WALL_SIZE = 24
 
-        self.hash = data[-4:]
-        self.sky_start = self.headerLen + self.header2Len
-        self.sky_length = (1<<self.pskybits) * 2
+        self.hash = self.data[-4:]
+        self.sky_start = self.HEADER_SIZE + self.header2Len
+        self.sky_length = (1 << self.pskybits) * 2
         self.sectors_start = self.sky_start + self.sky_length
+        return self.sectors_start
 
-    def WriteNumSprites(self, num):
-        self.num_sprites = num
+    def WriteNumSprites(self, pos):
```

Comment:
Blood doesn't use WriteNumSprites, maybe this function should just raise an exception

Reply:
We read that value in the head don't we? How does it get back into the file during serialisation?

Reply:
oh wait, the same way it got there to start with, using the header packer

Reply:
Hold on a sec.. there was an original impl for this method... it's not being used in master, ie before these changes?

Reply:
actually we do need to do `self.num_sprites = len(self.sprites)`